pkg/api/types.go

```diff
@@ -2143,6 +2143,20 @@ const (
 
 	// SSHAuthPrivateKey is the key of the required SSH private key for SecretTypeSSHAuth secrets
 	SSHAuthPrivateKey = "ssh-privatekey"
+
+	// SecretTypeTLS contains information about SSL termination used with the Ingress resource.
+	//
+	// Required fields:
+	// - Secret.Data["key"] - TLS private key.
+	//   Secret.Data["cert"] - TLS certificate.
+	//   Secret.Data["chain"] - optional certificate chain.
+	// TODO: Consider supporting different formats, specifying CA/destinationCA.
+	SecretTypeTLS SecretType = "kubernetes.io/tls"
```

Comment:
Is this secret the same no matter whether it's a client or server cert?  Should we have one type for client and one type for server?  I may have missed that discussion earlier, but it's an important distinction and we should call it out.